<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1121</width>
    <height>867</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="3" rowspan="3">
     <widget class="QGraphicsView" name="scene_graphicsView"/>
    </item>
    <item row="0" column="0" rowspan="3" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Алгоритмы построения окружностей/эллипсов</string>
      </property>
      <widget class="QPushButton" name="canon_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>271</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Каноническое уравнение</string>
       </property>
      </widget>
      <widget class="QPushButton" name="param_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>57</y>
         <width>271</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Параметрическое уравнение</string>
       </property>
      </widget>
      <widget class="QPushButton" name="bresenham_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>271</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Алгоритм Брезенхема</string>
       </property>
      </widget>
      <widget class="QPushButton" name="miidle_point_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>271</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Алгоритм средней точки</string>
       </property>
      </widget>
      <widget class="QPushButton" name="library_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>170</y>
         <width>271</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Библиотечный</string>
       </property>
      </widget>
      <widget class="QPushButton" name="line_color_pushButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>240</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Цвет линии</string>
       </property>
      </widget>
      <widget class="QPushButton" name="background_color_pushButton">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>240</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Цвет фона</string>
       </property>
      </widget>
      <widget class="QLabel" name="line_color_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>280</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="background_color_label">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>280</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="color_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>210</y>
         <width>261</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Выбор цвета&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>310</y>
         <width>295</width>
         <height>481</height>
        </rect>
       </property>
       <property name="title">
        <string>Ввод параметров окружности/эллипса</string>
       </property>
       <widget class="QLabel" name="x_center_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>111</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Х центра</string>
        </property>
       </widget>
       <widget class="QLabel" name="y_center_label">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>20</y>
          <width>131</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Y центра</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="x_center_lineEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
        <property name="clearButtonEnabled">
         <bool>false</bool>
        </property>
       </widget>
       <widget class="QLineEdit" name="y_center_lineEdit">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>40</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="circle_radius_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>R окружности</string>
        </property>
       </widget>
       <widget class="QLabel" name="width_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>90</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Ширина эллипса Rx</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="circle_radius_lineEdit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>60</y>
          <width>131</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="width_lineEdit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>90</y>
          <width>131</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="height_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>110</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Высота эллипса Ry</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="height_lineEdit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>110</y>
          <width>131</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="add_circle_pushButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>140</y>
          <width>141</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Построить окружность</string>
        </property>
       </widget>
       <widget class="QPushButton" name="add_ellipse_pushButton">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>140</y>
          <width>121</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Построить эллипс</string>
        </property>
       </widget>
       <widget class="QLabel" name="start_radius_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>190</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Начальный радиус</string>
        </property>
       </widget>
       <widget class="QLabel" name="end_radius_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>230</y>
          <width>111</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Конечный радиус</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="start_radius_lineEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>210</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="end_radius_lineEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>250</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="step_circle_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>270</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Шаг радиуса</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="step_circle_lineEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>290</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="count_circles_label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>310</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Число окружностей</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="count_circles_lineEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>330</y>
          <width>101</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="circle_spectrum_pushButton">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>380</y>
          <width>131</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Спектр окружностей</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="start_width_lineEdit">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>210</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="start_width_label">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>190</y>
          <width>141</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Начальная ширина Rx</string>
        </property>
       </widget>
       <widget class="QLabel" name="start_height_label">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>230</y>
          <width>131</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Начальная высота Ry</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="start_height_lineEdit">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>250</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="step_ellipse_label">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>270</y>
          <width>141</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Шаг изменения полуоси</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="step_ellipse_lineEdit">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>290</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="count_ellipse_label">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>340</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Число эллипсов</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="count_ellipse_lineEdit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>360</y>
          <width>131</width>
          <height>16</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="ellipse_spectrum_pushButton">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>380</y>
          <width>141</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Спектр эллипсов</string>
        </property>
       </widget>
       <widget class="QPushButton" name="rx_pushButton">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>310</y>
          <width>71</width>
          <height>28</height>
         </rect>
        </property>
        <property name="text">
         <string>Rx</string>
        </property>
       </widget>
       <widget class="QPushButton" name="ry_pushButton">
        <property name="geometry">
         <rect>
          <x>220</x>
          <y>310</y>
          <width>61</width>
          <height>28</height>
         </rect>
        </property>
        <property name="text">
         <string>Ry</string>
        </property>
       </widget>
       <widget class="QPushButton" name="circle_time_pushButton">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>420</y>
          <width>131</width>
          <height>61</height>
         </rect>
        </property>
        <property name="text">
         <string>Временные 
характеристики
окружностей</string>
        </property>
       </widget>
       <widget class="QPushButton" name="ellipse_time_pushButton">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>420</y>
          <width>131</width>
          <height>61</height>
         </rect>
        </property>
        <property name="text">
         <string>Временные 
характеристики
эллипсов</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1121</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="program_menu">
    <property name="title">
     <string>О программе</string>
    </property>
   </widget>
   <widget class="QMenu" name="author_menu">
    <property name="title">
     <string>Об авторе</string>
    </property>
   </widget>
   <widget class="QMenu" name="exit_menu">
    <property name="title">
     <string>Выход</string>
    </property>
   </widget>
   <widget class="QMenu" name="clear_menu">
    <property name="title">
     <string>Очистка экрана</string>
    </property>
   </widget>
   <addaction name="program_menu"/>
   <addaction name="author_menu"/>
   <addaction name="exit_menu"/>
   <addaction name="clear_menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
